src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,348 @@
+          </svg>
+        );
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            iconSvg: icon,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(getByTestId(iconTestId)).toBeTruthy();
+      });
+
+      test.each([true, false])('Disables click events when disabled prop is %s', disabledValue => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const mockClickSpy = jest.fn();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            disabled: disabledValue,
+            onClick: mockClickSpy,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button')!;
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        button.click();
+        expect(mockClickSpy).toBeCalledTimes(disabledValue ? 0 : 1);
+      });
+
+      test('renders an empty button when no iconName and iconSVG prop', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper } = renderVisualRefreshTriggerButton(
+          {
+            iconName: '' as IconProps.Name,
+            iconSvg: '',
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(button).toBeTruthy();
+        expect(button?.findIcon()).toBeNull();
+      });
+    });
+  });
+
+  test('renders correctly with badge using dot class on desktop', () => {
+    const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+    const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+      {
+        badge: true,
+      },
+      {
+        isMobile: false,
+      },
+      ref
+    );
+
+    expect(wrapper).not.toBeNull();
+    const button = wrapper.find('button');
+    expect(getByTestId(mockTestId)).toBeTruthy();
+    expect(button).toBeTruthy();
+    expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+    expect(wrapper!.findIcon()).toBeTruthy();
+    expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeTruthy();
+  });
+
+  test.each([true, false] as const)('Is focusable using the forwarded ref with mobile is %s', isMobile => {
+    const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
```

Comment:
why is this type ButtonProps.Ref | null? In this case, it should be guaranteed that this ref is always assigned, shouldn't it?